application/models/entry_models.py

```diff
@@ -0,0 +1,226 @@
+    
+    """
+    def __init__(self):
+
+        self.title = ''
+        self.body = '' 
+        self.date_modified = None
+        self.date_created = datetime.date.today()
+	    
+        
+
+    def save(self, current_user_email):
+        # insert data into db
+        query = "INSERT INTO entries (owner_id, title, body, date_created, date_modified) \
+                VALUES ((SELECT user_id from users where email ='{}'), '{}', '{}', '{}','{}')" \
+                                                    . format(current_user_email,
+                                                        self.title,
+                                                        self.body,
+                                                        self.date_created,
+                                                        self.date_modified
+                                                        )
+        db.execute(query)        
+
+
+    """ get a single diary entry """
+   
+    def get_diary_entry(self, entry_id=None):
+        
+  
+        query = "SELECT * from entries where entry_id = {}"\
+                . format(entry_id)      
+        result = db.execute(query)
+        row = result.fetchone()
+        
+        return [{'id': row[0], 'title': row[2],'body': row[3], \
+                                            'date_created' : row[4],'date_modified' : row[5]}], 200
+        
+    
+    """ update a diary entry """
+    def update_diary_entry(self,entry_id):
+        """ get diary entry by id """
+        query = "select * from entries where entry_id='{}'".format(entry_id)
+        result = db.execute(query)
+        entry = result.fetchone()
+
+
+        """ if not found """
+        if entry is None:
+            return {'message': 'diary entry with given id does not exist'}, 404
+        current_user_email = get_jwt_identity()
+        query =  "select user_id from users where email='{}'"\
+                    . format(current_user_email)
+        result = db.execute(query)
+        user_id = result.fetchone()
+        """ if entry owner id doesnot match id of current user """
+        if not entry[1] == user_id[0]:
+            return {'message': 'You cannot change \
+                        details of diary entry that you do not own'}, 401
+        try:
+
+            data = request.get_json()
+
+            title_json = data['title']
+            body_json = data['body']
+
+            title = title_json.strip()
+            body = body_json.strip()
+
+            """ validate for duplicate entry titles """
+            
+            query = "select * from entries where title='{}'".format(title)
+            result = db.execute(query)
+            rows = result.fetchall() 
+            if(len(rows) > 0):
+                return {'message': 'diary entry with such title already exists'}, 406
+
+            """ validate for alphanumeric characters """
+
+            if not re.match('^[a-zA-Z0-9_]+$',title):
```

Comment:
You regular expression is not handling scenarios where the title contains spaces ie more than one word or that was your design thought?